JetNews/app/src/main/java/com/example/jetnews/utils/WindowSize.kt

```diff
@@ -39,50 +38,44 @@ import androidx.window.layout.WindowMetricsCalculator
  */
 enum class WindowSize { Compact, Medium, Expanded }
 
+/**
+ * Partitions a [DpSize] into a enumerated [WindowSize] class.
+ */
 @VisibleForTesting
-fun getWindowSize(width: Dp): WindowSize = when {
-    width.value < 0f -> throw IllegalArgumentException("Dp value cannot be negative")
-    width.value < 600f -> WindowSize.Compact
-    width.value < 840f -> WindowSize.Medium
+fun getWindowSizeClass(windowDpSize: DpSize): WindowSize = when {
+    windowDpSize.width < 0.dp -> throw IllegalArgumentException("Dp value cannot be negative")
+    windowDpSize.width < 600.dp -> WindowSize.Compact
+    windowDpSize.width < 840.dp -> WindowSize.Medium
     else -> WindowSize.Expanded
 }
 
 /**
- * Remembers the [WindowSize] corresponding to the current window metrics.
+ * Remembers the [Size] in pixels of the window corresponding to the current window metrics.
  */
 @Composable
-fun rememberWindowSizeState(): WindowSize {
-    val windowMetrics = rememberCurrentWindowMetrics()
-    val density = LocalDensity.current
-
-    return remember(windowMetrics, density) {
-        with(density) {
-            getWindowSize(windowMetrics.bounds.width().toDp())
-        }
+private fun Activity.rememberWindowSize(): Size {
+    val configuration = LocalConfiguration.current
+    // WindowMetricsCalculator implicitly depends on the configuration through the activity,
+    // so recalculate it upon changes.
+    val windowMetrics = remember(configuration) {
+        WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(this)
     }
+    return windowMetrics.bounds.toComposeRect().size
 }
 
 /**
- * Returns the [WindowMetrics] corresponding to the current activities
- * [WindowMetricsCalculator.computeCurrentWindowMetrics].
+ * Remembers the [WindowSize] class for the window corresponding to the current window metrics.
  */
 @Composable
-private fun rememberCurrentWindowMetrics(): WindowMetrics {
-    val activity = LocalContext.current.findActivity()
-    val configuration = LocalConfiguration.current
-    return remember(activity, configuration) {
-        WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(activity)
-    }
-}
+fun Activity.rememberWindowSizeClass(): WindowSize {
```

Comment:
Nit: Move this public function above the private one

Reply:
Done!